src/stores/ThreepidInviteStore.ts

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import EventEmitter from "events";
+import { base32 } from "rfc4648";
+
+// Dev note: the interface is split in two so we don't have to disable the
+// linter across the whole project.
+export interface IThreepidInviteWireFormat {
+    email: string;
+    signurl: string;
+    room_name: string; // eslint-disable-line camelcase
+    room_avatar_url: string; // eslint-disable-line camelcase
+    inviter_name: string; // eslint-disable-line camelcase
+
+    // TODO: Figure out if these are ever populated
+    guest_access_token?: string; // eslint-disable-line camelcase
+    guest_user_id?: string; // eslint-disable-line camelcase
+}
+
+interface IPersistedThreepidInvite extends IThreepidInviteWireFormat {
+    roomId: string;
+}
+
+export interface IThreepidInvite {
+    id: string; // generated by us
+    roomId: string;
+    toEmail: string;
+    signUrl: string;
+    roomName: string;
+    roomAvatarUrl: string;
+    inviterName: string;
+}
+
+const STORAGE_PREFIX = "mx_threepid_invite_";
+
+export default class ThreepidInviteStore extends EventEmitter {
+    private static _instance: ThreepidInviteStore;
+
+    public static get instance(): ThreepidInviteStore {
+        if (!ThreepidInviteStore._instance) {
+            ThreepidInviteStore._instance = new ThreepidInviteStore();
+        }
+        return ThreepidInviteStore._instance;
+    }
+
+    public storeInvite(roomId: string, wireInvite: IThreepidInviteWireFormat): IThreepidInvite {
+        const invite = <IPersistedThreepidInvite>{roomId, ...wireInvite};
+        const id = this.generateIdOf(invite);
+        localStorage.setItem(`${STORAGE_PREFIX}${id}`, JSON.stringify(invite));
+        return this.translateInvite(invite);
+    }
+
+    public getWireInvites(): IPersistedThreepidInvite[] {
+        const results: IPersistedThreepidInvite[] = [];
+        for (let i = 0; i < localStorage.length; i++) {
+            const keyName = localStorage.key(i);
+            if (!keyName.startsWith(STORAGE_PREFIX)) continue;
+            results.push(JSON.parse(localStorage.getItem(keyName)) as IPersistedThreepidInvite);
+        }
+        return results;
+    }
+
+    public getInvites(): IThreepidInvite[] {
```

Comment:
this is optimistic for https://github.com/vector-im/element-web/issues/15134 (as is a lot of this store behaviour)

The eventual plan is to drop `pickBestInvite()`, `getWireInvites()`, and `translateToWireFormat()`, but for now they are the fastest path forwards with the time available.